packages/minifier-css/minifier.js

```diff
@@ -1,112 +1,209 @@
-    };
+    // located in the beginning of a file. Before concatenation, pull
+    // @import rules to the beginning of a new syntax tree so they always
+    // precede other rules.
+    const newAst = postcss.root();
 
-    _.each(cssAsts, function (ast) {
-      // Pick only the imports from the beginning of file ignoring @charset
-      // rules as every file is assumed to be in UTF-8.
-      var charsetRules = _.filter(ast.stylesheet.rules,
-                                  rulesPredicate("charset"));
-
-      if (_.any(charsetRules, function (rule) {
-        // According to MDN, only 'UTF-8' and "UTF-8" are the correct encoding
-        // directives representing UTF-8.
-        return ! /^(['"])UTF-8\1$/.test(rule.charset);
-      })) {
-        warnCb(ast.filename, "@charset rules in this file will be ignored as UTF-8 is the only encoding supported");
-      }
+    cssAsts.forEach((ast) => {
+      if (ast.nodes) {
+        // Pick only the imports from the beginning of file ignoring @charset
+        // rules as every file is assumed to be in UTF-8.
+        const charsetRules = ast.nodes.filter(rulesPredicate('charset'));
 
-      ast.stylesheet.rules = _.reject(ast.stylesheet.rules,
-                                      rulesPredicate("charset"));
-      var importCount = 0;
-      for (var i = 0; i < ast.stylesheet.rules.length; i++)
-        if (! rulesPredicate(["import", "comment"])(ast.stylesheet.rules[i])) {
-          importCount = i;
-          break;
+        if (charsetRules.some((rule) => {
+          // According to MDN, only 'UTF-8' and "UTF-8" are the correct
+          // encoding directives representing UTF-8.
+          return ! /^(['"])UTF-8\1$/.test(rule.params);
+        })) {
+          warnCb(
+            ast.filename,
+            '@charset rules in this file will be ignored as UTF-8 is the ' +
+            'only encoding supported'
+          );
         }
 
-      CssTools.rewriteCssUrls(ast);
+        ast.nodes = ast.nodes.filter(rulesPredicate('charset', true));
+        let importCount = 0;
+        for (let i = 0; i < ast.nodes.length; i++) {
+          if (! rulesPredicate(['import', 'comment'])(ast.nodes[i])) {
+            importCount = i;
+            break;
+          }
+        }
 
-      var imports = ast.stylesheet.rules.splice(0, importCount);
-      newAst.stylesheet.rules = newAst.stylesheet.rules.concat(imports);
+        CssTools.rewriteCssUrls(ast);
 
-      // if there are imports left in the middle of file, warn user as it might
-      // be a potential bug (imports are valid only in the beginning of file).
-      if (_.any(ast.stylesheet.rules, rulesPredicate("import"))) {
-        // XXX make this an error?
-        warnCb(ast.filename, "there are some @import rules those are not taking effect as they are required to be in the beginning of the file");
-      }
+        const imports = ast.nodes.splice(0, importCount);
+        newAst.nodes = newAst.nodes.concat(imports);
```

Comment:
Any reason not to do `newAst.nodes.push(...imports)` here?

Reply:
No reason at all - I'll update, thanks!